src/cpp/include/openvino/genai/text2image/pipeline.hpp

```diff
@@ -0,0 +1,165 @@
+                                                      Type scheduler_type = AUTO);
+
+        virtual ~Scheduler();
+    };
+
+    struct OPENVINO_GENAI_EXPORTS GenerationConfig {
+        // LCM: prompt only w/o negative prompt
+        // SD XL: prompt2 and negative_prompt2
+        // FLUX: prompt2 (prompt if prompt2 is not defined explicitly)
+        // SD 3: prompt2, prompt3 (with fallback to prompt) and negative_prompt2, negative_prompt3
+        std::string prompt2, prompt3;
+        std::string negative_prompt, negative_prompt2, negative_prompt3;
+
+        size_t num_images_per_prompt = 1;
+
+        // random generator to have deterministic results
+        std::shared_ptr<Generator> random_generator = std::make_shared<CppStdGenerator>(42);
+
+        // the following values depend on HF diffusers class used to perform generation
+        float guidance_scale = 7.5f;
+        int64_t height = -1;
+        int64_t width = -1;
+        size_t num_inference_steps = 50;
+
+        void update_generation_config(const ov::AnyMap& config_map);
+
+        // checks whether is config is valid
+        void validate() const;
+
+        template <typename... Properties>
+        ov::util::EnableIfAllStringAny<void, Properties...> update_generation_config(Properties&&... properties) {
+            return update_generation_config(ov::AnyMap{std::forward<Properties>(properties)...});
+        }
+    };
+
+    explicit Text2ImagePipeline(const std::string& root_dir);
+
+    Text2ImagePipeline(const std::string& root_dir, const std::string& device, const ov::AnyMap& properties = {});
+
+    // creates either LCM or SD pipeline from building blocks
+    static Text2ImagePipeline stable_diffusion(
+        const std::shared_ptr<Scheduler>& scheduler_type,
+        const CLIPTextModel& clip_text_model,
+        const UNet2DConditionModel& unet,
+        const AutoencoderKL& vae_decoder);
+
+    // creates either LCM or SD pipeline from building blocks
+    static Text2ImagePipeline latent_consistency_model(
+        const std::shared_ptr<Scheduler>& scheduler_type,
+        const CLIPTextModel& clip_text_model,
+        const UNet2DConditionModel& unet,
+        const AutoencoderKL& vae_decoder);
+
+    GenerationConfig get_generation_config() const;
+    void set_generation_config(const GenerationConfig& generation_config);
+
+    // ability to override scheduler
+    void set_scheduler(std::shared_ptr<Scheduler> scheduler);
+
+    // with static shapes performance is better
+    void reshape(const int num_images_per_prompt, const int height, const int width, const float guidance_scale);
+
+    void compile(const std::string& device, const ov::AnyMap& properties = {});
+
+    // Returns a tensor with the following dimensions [num_images_per_prompt, height, width, 3]
+    ov::Tensor generate(const std::string& positive_prompt, const ov::AnyMap& properties = {});
```

Comment:
Is providing intermediate results something that is needed? To stich them into a GIF, for example. I guess that could be one extra config option and one extra dimension in the returned tensor.

Reply:
what are intermediate results? I suppose NHWC is kind of standard which can be used everywhere

Reply:
It is possible to save intermediate images in latent space and make images out of them as well.

Reply:
but we don't need this mode for now, right?

Reply:
That's my question :)
> Is providing intermediate results something that is needed?

Reply:
I think no
Current version provides the same results as HF.
New modes can be enabled in future as per customers requests.